src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -268,16 +279,15 @@ static File makeChunkFile(File baseDir, int i)
   public static class Outer implements SmooshedWriter
   {
     private final int fileNum;
-    private final OutputStream out;
     private final int maxLength;
+    private final GatheringByteChannel channel;
 
-    private boolean open = true;
     private int currOffset = 0;
 
-    Outer(int fileNum, OutputStream out, int maxLength)
+    Outer(int fileNum, FileChannel channel, int maxLength)
     {
       this.fileNum = fileNum;
-      this.out = out;
+      this.channel = new BufferedWritableByteChannel(channel);
```

Comment:
@leventov Sorry for delay. The test throws exception and does not flush or close underlying stream, making difference in file size. If it's acceptable change by metamx, I'm really happy on removing buffering channel.